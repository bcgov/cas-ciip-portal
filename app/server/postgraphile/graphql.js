const {
  createPostGraphileSchema,
  withPostGraphileContext,
} = require("postgraphile");
const { graphql } = require("graphql");
const authenticationPgSettings = require("./authenticationPgSettings");
const { pgPool, getDatabaseUrl } = require("../db/databaseConnectionService");
const postgraphileOptions = require("./postgraphileOptions");

const pgSettings = (req) => {
  const opts = {
    ...authenticationPgSettings(req),
  };
  return opts;
};

let postgraphileSchemaSingleton;

const postgraphileSchema = async () => {
  if (!postgraphileSchemaSingleton) {
    postgraphileSchemaSingleton = await createPostGraphileSchema(
      getDatabaseUrl(),
      process.env.DATABASE_SCHEMA || "ggircs_portal",
      postgraphileOptions()
    );
  }

  return postgraphileSchemaSingleton;
};

async function performQuery(query, variables, request) {
  const settings = pgSettings(request);

  return withPostGraphileContext(
    {
      pgPool,
      pgSettings: settings,
    },
    async (context) => {
      // `context` object, which should NOT be used outside of this
      // function.
      const result = graphql(
        await postgraphileSchema(),
        query,
        null,
        { ...context }, // You can add more to context if you like
        variables
      );
      return result;
    }
  );
}

module.exports = {
  performQuery,
};
