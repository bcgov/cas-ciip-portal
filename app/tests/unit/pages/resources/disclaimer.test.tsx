import React from 'react';
import {shallow} from 'enzyme';
import Disclaimer from 'pages/resources/disclaimer';
import {disclaimerQueryResponse} from 'disclaimerQuery.graphql';

const query: disclaimerQueryResponse['query'] = {
  session: {
    ' $fragmentRefs': {
      defaultLayout_session: true
    }
  }
};

const router = {
  query: {
    applicationId: 'skjdh839',
    hasSwrsReport: true,
    version: 1
  }
};

describe('BCGov core disclaimer page', () => {
  it('It matches the last accepted Snapshot', () => {
    const wrapper = shallow(
      <Disclaimer query={query} router={router} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
