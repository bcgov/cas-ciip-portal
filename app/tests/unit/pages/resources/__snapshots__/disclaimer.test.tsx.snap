// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BCGov core disclaimer page It matches the last accepted Snapshot 1`] = `
<Relay(DefaultLayout)
  needsSession={false}
  needsUser={false}
  session={
    Object {
      " $fragmentRefs": Object {
        "defaultLayout_session": true,
      },
    }
  }
  title="Disclaimer"
>
  <Container
    fluid={false}
  >
    <p>
      <strong>
        Warranty Disclaimer
      </strong>
    </p>
    <p>
      This information is provided as a public service by the Government of British Columbia, Box 9411, Victoria, British Columbia, Canada V8W 9V1.
    </p>
    <p>
      This website and all of the information it contains are provided "as is" without warranty of any kind, whether express or implied. All implied warranties, including, without limitation, implied warranties of merchantability, fitness for a particular purpose, and non-infringement, are hereby expressly disclaimed. Links and references to any other websites are provided for information only and listing shall not be taken as endorsement of any kind. The Government of British Columbia is not responsible for the content or reliability of the linked websites and does not endorse the content, products, services or views expressed within them.
    </p>
    <p>
      <strong>
        Limitation of Liabilities
      </strong>
    </p>
    <p>
      Under no circumstances will the Government of British Columbia be liable to any person or business entity for any direct, indirect, special, incidental, consequential, or other damages based on any use of this website or any other website to which this site is linked, including, without limitation, any lost profits, business interruption, or loss of programs or information, even if the Government of British Columbia has been specifically advised of the possibility of such damages.
    </p>
  </Container>
</Relay(DefaultLayout)>
`;
